doc/md/canister-maintenance/upgrades.md

```diff
@@ -207,9 +205,11 @@ For more information, see the [example of explicit migration](compatibility.md#e
 
 The following aspects are retained for historical reasons and backwards compatibility:
 
-### Preupgrade and postupgrade system methods
+### Pre-upgrade and post-upgrade system methods
 
-This is an advanced functionality that is not recommended for standard cases, as it is error-prone and can render the canister unusable.
+:::danger
+Using the pre- and post-upgrade system methods is discouraged. It is error-prone and can render a canister unusable. Per best practices, using these methods should be avoided if possible.
```

Comment:
```suggestion
Using the pre- and post-upgrade system methods is discouraged. It is error-prone and can render a canister unusable. In particular, if a `preupgrade` method traps and cannot be prevented from trapping by other means, then your canister may be left in a state in which it can no longer be upgraded.  Per best practices, using these methods should be avoided if possible.
```